.dockerignore

```diff
@@ -0,0 +1,15 @@
+#Ignore the following directories
+backend/build/
+backend/.gradle/
+frontend/build/
+frontend/node_modules/
+nginx/
+ops/
```

Comment:
Can we please remove the exclusion of the `ops` directory? We do build images there.

Reply:
While I don't love the addition of this file because I don't particularly like that we're building the `backend` and `frontend` images in the root of the project, ideally, we should build those projects in their respective directories. The changes this PR makes, from setting volume mounts to copying it in during builds, is another good approach to getting the `.git` directory into the container, and because that's a requirement at the moment, I don't have a better solution without more changes.

This `.dockerignore` will only impact builds where the docker context is the project's root, so it shouldn't impact builds in other directories. As far as I can tell, the context for the DB-client builds takes place in `./ops/services/container_instances/db_client/image`. If that's true, this `ops/` inclusion should be fine.

Reply:
yeah the .dockerignore will only affect the builds that have the context set as the repo root (only frontend and backend right now since the git situation)
But nonetheless, I went ahead and made it to be more targeted and changed the 
`/ops/`
to 
`ops/services/app_functions/report_stream_batched_publisher/functions/node_modules/`